pandas/core/frame.py

```diff
@@ -6960,8 +6965,11 @@ def explode(self, column: Union[str, Tuple]) -> "DataFrame":
         # TODO: use overload to refine return type of reset_index
         assert df is not None  # needed for mypy
         result = df[column].explode()
-        result = df.drop([column], axis=1).join(result)
-        result.index = self.index.take(result.index)
+        if ignore_index:
+            result = df.drop([column], axis=1).join(result).reset_index(drop=True)
```

Comment:
You can assign the ``.index`` a default index to avoid the additional copy in `reset_index`, in a similar way as done in eg https://github.com/pandas-dev/pandas/pull/30405

Reply:
Thank you for the suggestion, can you have a look if I implemented what you ment?